src/Components/LoginPage/Signup.tsx

```diff
@@ -0,0 +1,58 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import {
+  FormContainer,
+  PasswordElement,
+  TextFieldElement,
+} from "react-hook-form-mui";
+import { Button, Stack, Box } from "@mui/material";
+import { createUser } from "~/Helper Functions/ApiCalls";
+import { Link as RouterLink } from "react-router";
+import signupSchema from "~/types/Forms/SignupSchema";
+import SignupFormInputs from "~/types/Forms/SignupFormInputs";
+
+export default function SignupView() {
+  const formResolver = zodResolver(signupSchema);
+  const formContext = useForm<SignupFormInputs>({ resolver: formResolver });
+  const { handleSubmit, setError } = formContext;
+
+  const onSubmit = async (data: SignupFormInputs) => {
+    createUser(data)
+      .then((response) => {
+        response.json().then((data) => {
+          alert(data.message);
```

Comment:
```suggestion
          alert(data.message);
          setUserData(data);
```
It would be great if the data that comes back can be stored in the userData useState in App.tsx

Reply:
This should happen after login right ?

Reply:
See the comment either in this file